src/main/scala/inox/ast/SymbolOps.scala

```diff
@@ -1073,7 +1075,7 @@ trait SymbolOps { self: TypeOps =>
 
       def replace(path: Path, oldE: Expr, newE: Expr, body: Expr): Expr = transformWithPC(body) {
         (e, env, op) =>
-          if ((path.bindings.toSet subsetOf env.bindings.toSet) &&
+          if ((path.bindings.toSet subsetOf env.bindings.toSet) && // FIXME bindings or bounds?
```

Comment:
Probably both. You can probably add the condition `path.bounds subsetOf env.bounds` as well.